### ISSUE TITLE
Data from `set_stats` not getting saved to job `artifacts` field as a flaky occurrence

### ISSUE THREAD

**Issue by AlanCoding** [ansible/awx#11821]
### Please confirm the following

- [X] I agree to follow this project's [code of conduct](https://docs.ansible.com/ansible/latest/community/code_of_conduct.html).
- [X] I have checked the [current issues](https://github.com/ansible/awx/issues) for duplicates.
- [X] I understand that AWX is open source software provided for free and that I might not receive a timely response.

### Summary

Sometimes the job's `artifacts` field of a job is not getting populated with data from `set_stats` from within a playbook.

The playbook content I'm using to test this comes from https://github.com/ansible/test-playbooks/pull/153

After running that and confirming the job is fully finished, I find that artifacts is `{}`.

Nothing relevant appears in the logs.

This happens in clusters and control-standalone scenarios. Those are the only ones which have been tested.

### AWX version

devel

### Select the relevant components

- [ ] UI
- [X] API
- [ ] Docs

### Installation method

docker on linux

### Modifications

no

### Ansible version

2.12

### Operating system

Fedora / centos / RHEL

### Web browser

Chrome

### Steps to reproduce

```python
        jt = factories.job_template(
            project=project_from_PR(153, 'Playbook to put environment in artifacts'),
            playbook='print_environment.yml',
            allow_simultaneous=True
        )
        jt.ds.inventory.add_host()
        jobs = [jt.launch() for i in range(16)]
        for job in jobs:
            job.wait_until_completed()
            assert job.artifacts, f'Failed to get artifacts from job {job.id}, output: {job.result_stdout}'
            assert 'printenv' in job.artifacts
```

To summarize, use the test-playbooks content to make a project, create a JT with allow_simultaneous, launch 16 jobs right away. After they have all been launched, wait for completion. After they complete, inspect artifacts field

### Expected results

The playbook content will populate the artifacts with the discovered environment variables and other stuff.

### Actual results

In job details page, `/api/v2/jobs/N/`

```json
    "artifacts": {},
```

**Comment by AlanCoding** [comment]
I re-ran the script here and was not able to detect any failures, so it seems this was fixed by some unknown changes.
